packages/amplify-cli-core/src/types.ts

```diff
@@ -401,3 +599,13 @@ interface AmplifyToolkit {
   invokePluginMethod: <T>(context: $TSContext, category: string, service: string | undefined, method: string, args: $TSAny[]) => Promise<T>;
   getTags: (context: $TSContext) => Tag[];
 }
+export interface IOptionFlowHeadlessData {
+  input: string;
+  timestamp: number;
+}
+export interface IOptionFlowCLIData {
+  prompt: string;
+  input: unknown;
+  timestamp: number;
+}
+export type TypeOptionFlowData = IOptionFlowHeadlessData | IOptionFlowCLIData;
```

Comment:
Few things here.
- I'm on the fence regarding the `I` prefix for interfaces. In general I'd rather name implementations with more detailed name or use something for default impl like `DefaultFoo implements Foo` or `$Foo implements Foo`. However, we have plenty of existing interfaces with`I` so we should probably be consistent.
- I would definietly drop `Type` from `TypeOptionFlowData`
- why is `input` string in one shape and `unknown` in the other. Isn't this creating diamond problem in `TypeOptionFlowData` ?
- should `IOptionFlowCLIData` extend `IOptionFlowHeadlessData` ?

Reply:
- The `I` prefix is considered unidiomatic pretty widely in the Typescript world. Personally, I dislike it, as it encourages a somewhat lazy naming pattern, and also implicitly endorses the notion that an interface is just a class with its guts ripped out. I think they should be removed. I can make that change in this PR.
- Good points on the shapes of the types. I copied those verbatim from `cli`. Let me take a closer look at them and see if I can narrow them down.

Reply:
I opted against making the change in this PR. I will handle renaming these in a follow-up. The PR has grown quite large, and this would make reviewing the import changes more difficult for reviewers.

Reply:
Sounds good. Could you record refactorings that you leave for follow up in github issue or asana task, so that we don't lose them.